test/yapi_network_test.rb

```diff
@@ -0,0 +1,374 @@
+#! /usr/bin/env rspec
+
+require_relative "test_helper"
+
+require "yast"
+
+Yast.import "YaPI::NETWORK"
+Yast.import "NetworkInterfaces"
+Yast.import "Routing"
+Yast.import "IP"
+Yast.import "Service"
+Yast.import "DNS"
+Yast.import "LanItems"
+
+describe Yast::YaPI::NETWORK do
+  include YaPINetworkStub
+
+  describe ".Write" do
+
+    context "with routes" do
+      # Mock the system to do not break anything
+      before do
+        allow(Yast::Routing).to receive(:Read)
+        allow(Yast::Routing).to receive(:Routes)
+        allow(Yast::Routing).to receive(:Write)
+      end
+
+      subject { Yast::YaPI::NETWORK.Write("route" => route) }
+
+      let(:success) { { "error" => "", "exit" => "0" } }
+
+      context "with no gateway in the default route" do
+        let(:route) { { "default" => nil } }
+
+        it "returns success" do
+          expect(subject).to eq success
+        end
+
+        it "empties the routes" do
+          expect(Yast::Routing).to receive(:Routes=).with []
+          subject
+        end
+      end
+
+      context "with empty gateway in the default  route" do
+        let(:route) { { "default" => { "via" => "" } } }
+
+        it "returns success" do
+          expect(subject).to eq success
+        end
+
+        it "empties the routes" do
+          expect(Yast::Routing).to receive(:Routes=).with []
+          subject
```

Comment:
sure. But testing for ```=``` assumes that modification is done only by assignment. What I tried to suggest is to find a solution which should be more resistant to further refactoring, when updating ```Routes```can be done differently. Of course we must not rely on ```@Routes``` but relying on ```Routes``` is acceptable from my POV bcs it is a part of public API, a getter which can stay even in a future at least as an alias.